data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -482,14 +489,19 @@ public Builder withIsmPolicyFile(final String ismPolicyFile) {
             return this;
         }
 
-        public Builder withAction(final String action) {
-            // Removed validation because action may have expresions
+        public Builder withAction(final String action, final ExpressionEvaluator expressionEvaluator) {
+            checkArgument((EnumUtils.isValidEnumIgnoreCase(BulkAction.class, action) || JacksonEvent.isValidFormatExpressions(action, expressionEvaluator)), "action must be one of the following: " + BulkAction.values());
             this.action = action;
             return this;
         }
 
-        public Builder withActions(final List<Map<String, Object>> actions) {
-            // No validation done here because actions may have expresions
+        public Builder withActions(final List<Map<String, Object>> actions, final ExpressionEvaluator expressionEvaluator) {
+            for (final Map<String, Object> actionMap: actions) {
+                String action = (String)actionMap.get("type");
+                if (action != null) {
+                    checkArgument((EnumUtils.isValidEnumIgnoreCase(BulkAction.class, action) || JacksonEvent.isValidFormatExpressions(action, expressionEvaluator)), "action must be one of the following: " + BulkAction.values());
```

Comment:
Thanks for adding this! Would be nice to give a more direct exception for if there is an invalid expression vs some value that is not one of the enum values. Up to you if you think this is something to add